Jackson Reflectionfree Serializer with Raw Types

### Describe the bug

When using new reflection free serializer with raw types `ResteasyReactiveJacksonProcessor` fails with `[ERROR] Caused by: java.lang.IllegalArgumentException: Not a parameterized type!`

Issue is: 

```
    @GET
    @Path("raw-types")
    @Produces(MediaType.APPLICATION_JSON)
    public Map rawTypes() {
        return Map.of("key", "value");
    }
```

Works with `quarkus.rest.jackson.optimization.enable-reflection-free-serializers=false` set
Also works when specifying the return type correctly (`Map<String, String>`)



### Expected behavior

clean package works without errors

### Actual behavior

build fails, service does not start

### How to Reproduce?

1. Checkout project: https://github.com/janscheidegger/reflectionfree-reproducer
2. mvn clean package

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.37.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java
@@ -568,8 +568,14 @@ public class ResteasyReactiveJacksonProcessor {
         if (effectiveReturnType.name().equals(ResteasyReactiveDotNames.SET) ||
                 effectiveReturnType.name().equals(ResteasyReactiveDotNames.COLLECTION) ||
                 effectiveReturnType.name().equals(ResteasyReactiveDotNames.LIST)) {
+            if (effectiveReturnType.kind() != Type.Kind.PARAMETERIZED_TYPE) {
+                return null;
+            }
             effectiveReturnType = effectiveReturnType.asParameterizedType().arguments().get(0);
         } else if (effectiveReturnType.name().equals(ResteasyReactiveDotNames.MAP)) {
+            if (effectiveReturnType.kind() != Type.Kind.PARAMETERIZED_TYPE) {
+                return null;
+            }
             effectiveReturnType = effectiveReturnType.asParameterizedType().arguments().get(1);
         }
         return effectiveReturnType;

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/RawTypeReflectionFreeSerializerTest.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/RawTypeReflectionFreeSerializerTest.java
@@ -1,0 +1,91 @@
+package io.quarkus.resteasy.reactive.jackson.deployment.test;
+
+import static org.hamcrest.Matchers.is;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusExtensionTest;
+import io.restassured.RestAssured;
+
+public class RawTypeReflectionFreeSerializerTest {
+
+    @RegisterExtension
+    static QuarkusExtensionTest test = new QuarkusExtensionTest()
+            .setArchiveProducer(new Supplier<>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addClasses(RawTypeResource.class)
+                            .addAsResource(
+                                    new StringAsset(
+                                            "quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true\n"),
+                                    "application.properties");
+                }
+            });
+
+    @Test
+    public void testRawMap() {
+        RestAssured.get("/raw/map")
+                .then()
+                .statusCode(200)
+                .body("key", is("value"));
+    }
+
+    @Test
+    public void testRawList() {
+        RestAssured.get("/raw/list")
+                .then()
+                .statusCode(200)
+                .body("[0]", is("a"));
+    }
+
+    @Test
+    public void testRawSet() {
+        RestAssured.get("/raw/set")
+                .then()
+                .statusCode(200)
+                .body("[0]", is("x"));
+    }
+
+    @Path("/raw")
+    public static class RawTypeResource {
+
+        @SuppressWarnings("rawtypes")
+        @GET
+        @Path("/map")
+        @Produces(MediaType.APPLICATION_JSON)
+        public Map rawMap() {
+            return Map.of("key", "value");
+        }
+
+        @SuppressWarnings("rawtypes")
+        @GET
+        @Path("/list")
+        @Produces(MediaType.APPLICATION_JSON)
+        public List rawList() {
+            return List.of("a", "b", "c");
+        }
+
+        @SuppressWarnings("rawtypes")
+        @GET
+        @Path("/set")
+        @Produces(MediaType.APPLICATION_JSON)
+        public Set rawSet() {
+            return Set.of("x", "y");
+        }
+    }
+}

